week2/2-2/index.js

```diff
@@ -0,0 +1,208 @@
+const INIT_BALANCE = 0;
+const HISTORY_LIST = [
+    {
+        category: "쇼핑",
+        title: "신발 구매",
+        amount: 112000,
+        type: "loss"
+    },
+    {
+        category: "용돈",
+        title: "용돈이당!!!",
+        amount: 100000,
+        type: "income"
+    },
+    {
+        category: "식비",
+        title: "노티드 도넛",
+        amount: 10500,
+        type: "loss"
+    },
+    {
+        category: "용돈",
+        title: "또 용돈이당!!!",
+        amount: 200000,
+        type: "income"
+    }
+];
+
+const createItem = (isIncome, category, title, amount) => {
+    const historyAssetClass = isIncome ? "plus" : "minus";
+    const historyItemClass = isIncome ? "income" : "loss";
+    
+    const newItem = document.createElement("li");
+    newItem.classList.add("history_item");
+    newItem.classList.add(`${historyItemClass}`);
+    
+    newItem.innerHTML = `
+        <button type="button">
+            <img src="assets/x_icon.png" alt="x 버튼" class="x_icon">
+        </button>
+        <p class="history_category">${category}</p>
+        <p class="history_title">${title}</p>
+        <p class="history_asset ${historyAssetClass}">${isIncome ? "+" : "-"}${amount}</p>
+    `;
+    
+    const list = document.querySelector("ul");
+    list.appendChild(newItem);
+
+    const xButtons = document.querySelectorAll('.x_icon');
+
+    xButtons.forEach(button => {
+        button.addEventListener('click', function() {
+            const listItem = button.closest('.history_item');
+            if (listItem) {
+                listItem.remove();
+                updateAsset();
+            }
+        });
+    });
+}
+
+
+const initListItem = () => {
+
+    const totalAssetElement = document.querySelector('.total_asset');
+    totalAssetElement.textContent = INIT_BALANCE;
+
+    HISTORY_LIST.forEach(item => {
+        const isIncome = item.type === "income" ? true : false;
+        createItem(isIncome, item.category, item.title, item.amount);
+    });
```

Comment:
```suggestion
    HISTORY_LIST.forEach(({type, category, title, amount})=> {
        const isIncome = type === "income";
        createItem(isIncome, category, title, amount);
    });
```
요로코롬 구조분해할당하면 좀 더 깔끔 + isIncome은 그냥 삼항연산자 없이 이렇게 줘도 가능하겠죠 ! !